src/containers/Dashboard.js

```diff
@@ -1,42 +1,231 @@
 import React from "react";
+import { BrowserRouter as Router, Route, Link } from "react-router-dom";
 import produce from "immer";
+import * as R from "ramda";
 
 import apiClient from "../lib/apiClient";
+import { jwtDecode } from "../lib/localStorage";
+
+// Components
+import DashboardApiKeys from "./DashboardApiKeys";
+import DashboardStats from "./DashboardStats";
+import DashboardUserSettings from "./DashboardUserSettings";
+
+// Styles
+import "../styles/Dashboard.css";
+import "../styles/Buttons.css";
 
 export default class Dashboard extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      logoutError: ""
+      firstName: "",
+      lastName: "",
+      email: "",
+      userId: null,
+      apiKeys: [],
+      isLoading: true,
+      error: ""
     };
   }
 
+  componentDidMount() {
+    this.setState(
+      produce(draft => {
+        draft.isLoading = true;
+      })
+    );
+
+    const authToken = localStorage.getItem("cah-token");
+
+    if (!authToken) {
+      // TODO: check if token exists and handle
+    }
+
+    const decoded = jwtDecode(authToken);
+
+    if (decoded === null) {
+      this.setState(
+        produce(draft => {
+          draft.error = "Error fetching your information. Please refresh the page.";
```

Comment:
Would refreshing the page help if the JWT token is null? Wouldn't the user need to sign in again?